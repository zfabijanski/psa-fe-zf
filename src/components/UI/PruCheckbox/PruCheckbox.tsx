import React from "react";
import styled, { css, CSSProp } from "styled-components/macro";
import { useIntl } from "react-intl";
import { IModalData } from "../../../models/Modal";
import { Icon } from "../Icon";
import { ModalInfoIcon } from "../ModalInfoIcon";
import { ILabelProps } from "../Label";
import { ValidationInfo } from "../ValidationInfo";

export enum ECheckboxType {
  regular = "regular",
  gridItem = "gridItem",
  calcTableItem = "calcTableItem",
  checkboxList = "checkboxList",
}

export interface ICheckboxProps extends IModalData {
  disabled?: boolean;
  checked?: boolean;
  isInvalid?: boolean;
  className?: string;
  onChange?: (checked: boolean) => void;
  labelProps?: ILabelProps;
  onlyPresentation?: boolean;
  checkboxType?: keyof typeof ECheckboxType;
  hideValidationInfo?: boolean;
  name?: string;
  css?: CSSProp;
}

export const PruCheckbox: React.FC<ICheckboxProps> = (props) => {
  const intl = useIntl();

  return (
    <>
      <Wrapper
        checkboxType={props.checkboxType!}
        disabled={props.disabled}
        css={props.css}
      >
        <Input
          type="checkbox"
          disabled={props.disabled}
          checked={props.checked}
          name={props.name}
          onChange={(e) =>
            !props.disabled &&
            !props.onlyPresentation &&
            props.onChange?.(e.target.checked)
          }
        />
        <CheckMark
          role="checkbox"
          className={props.className}
          isInvalid={props.isInvalid}
          disabled={props.disabled}
          onClick={() =>
            !props.disabled &&
            !props.onlyPresentation &&
            props.onChange?.(!props.checked)
          }
          onlyPresentation={props.onlyPresentation}
          checked={props.checked}
        >
          <StyledIcon
            name="old-check-white"
            disabled={props.disabled}
            width={16}
            color="white100"
          />
        </CheckMark>

        {(props.labelProps?.labelTrKey || props.labelProps?.labelText) && (
          <CheckboxLabel
            disabled={props.disabled}
            onlyPresentation={props.onlyPresentation}
          >
            {props.labelProps.labelText ||
              (props.labelProps.labelTrKey
                ? intl.formatMessage({ id: props.labelProps.labelTrKey })
                : "")}
          </CheckboxLabel>
        )}

        {(props.labelProps?.modalContentTrKey || props.labelProps?.modalHtml) &&
          props.labelProps?.labelTrKey && (
            <StyledModalInfoIcon
              modalContentTrKey={props.labelProps.modalContentTrKey}
              modalContent={props.labelProps.modalContent}
              modalButtons={props.labelProps.modalButtons}
              modalHtml={props.labelProps.modalHtml}
              modalHtmlTrValues={props.labelProps.modalHtmlTrValues}
            />
          )}
      </Wrapper>
      {props.checkboxType !== ECheckboxType.checkboxList &&
        !props.hideValidationInfo && (
          <ValidationInfo
            validations={props.labelProps?.validationInfoTrKeys}
          />
        )}
    </>
  );
};

PruCheckbox.defaultProps = {
  checkboxType: ECheckboxType.regular,
};

const Wrapper = styled.div<{
  checkboxType: keyof typeof ECheckboxType;
  disabled?: boolean;
  css?: CSSProp;
}>`
  display: inline-flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;

  ${({ checkboxType }) =>
    checkboxType !== ECheckboxType.checkboxList &&
    css`
      position: relative;
    `};

  ${({ checkboxType, theme, disabled }) =>
    (checkboxType === ECheckboxType.gridItem ||
      checkboxType === ECheckboxType.calcTableItem) &&
    css`
      display: flex;
      border: 1px solid ${theme.newColors.gray30};
      background-color: ${disabled
        ? theme.newColors.gray10
        : theme.newColors.white100};
      height: 52px;
      padding: 0 14px;
    `};

  ${({ checkboxType }) =>
    checkboxType === ECheckboxType.gridItem &&
    css`
      margin-top: 24px;
    `};

  ${(props) => props.css};
`;

const CheckMark = styled.div<{
  disabled?: boolean;
  isInvalid?: boolean;
  onlyPresentation?: boolean;
  checked?: boolean;
}>`
  width: 24px;
  min-width: 24px;
  height: 24px;
  border: 1px solid;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 12px;

  &:before {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
  }

  cursor: ${({ disabled, onlyPresentation }) => {
    if (disabled) return "not-allowed";
    if (onlyPresentation) return "default";
    return "pointer";
  }};

  ${({ theme, disabled, onlyPresentation, checked, isInvalid }) => {
    if (!checked) {
      switch (true) {
        case disabled:
          return css`
            background-color: ${theme.newColors.gray10};
            border-color: ${theme.newColors.gray60};
          `;

        case isInvalid:
          return css`
            background-color: ${theme.newColors.primary05};
            border-color: ${theme.newColors.error};

            &:hover {
              background-color: ${theme.newColors.primary20};
              border-color: ${theme.newColors.primary100};
            }
          `;

        case onlyPresentation:
          return css`
            background-color: ${theme.newColors.primary05};
            border-color: ${theme.newColors.primary80};
          `;

        default:
          return css`
            background-color: ${theme.newColors.primary05};
            border-color: ${theme.newColors.primary80};

            &:hover {
              background-color: ${theme.newColors.primary20};
              border-color: ${theme.newColors.primary100};
            }
          `;
      }
    }

    switch (true) {
      case disabled:
        return css`
          background-color: ${theme.newColors.gray60};
          border-color: ${theme.newColors.gray60};
        `;

      case isInvalid:
        return css`
          background-color: ${theme.newColors.error};
          border-color: ${theme.newColors.error};
        `;

      default:
        return css`
          background-color: ${theme.newColors.gray100};
          border-color: ${theme.newColors.gray100};
        `;
    }
  }}
`;

const StyledIcon = styled(Icon)<{ disabled?: boolean }>`
  visibility: hidden;
`;

const Input = styled.input`
  display: none;

  &:checked + ${CheckMark} ${StyledIcon} {
    visibility: initial;
  }
`;

const CheckboxLabel = styled.span<{
  disabled?: boolean;
  onlyPresentation?: boolean;
}>`
  font-size: 16px;
  line-height: 20px;
  ${({ theme, disabled, onlyPresentation }) =>
    (disabled || onlyPresentation) &&
    css`
      color: ${theme.newColors.gray60};
    `};
`;

const StyledModalInfoIcon = styled(ModalInfoIcon)`
  margin-left: 10px;
  position: relative;
  z-index: 1;
`;

export default PruCheckbox;
