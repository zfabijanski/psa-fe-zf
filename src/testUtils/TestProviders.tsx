import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import { Normalize } from "styled-normalize";
import GlobalStyles from "theme/globalStyles";
import { theme } from "theme/theme";
import rootStore, { configureStore, rootReducer as reducer } from "AppStore";
import { BrowserRouter } from "react-router-dom";
import { Modal } from "components/UI/Modal";
import { IntlProvider } from "react-intl";
import { plMessages } from "translations/pl";
import I18nProvider from "I18nProvider";

interface TestProvidersProps {
  store?: typeof rootStore;
  renderModal?: boolean;
  realIntlProvider?: boolean;
}

export const addTestProviders =
  ({
    store = configureStore({ reducer }),
    renderModal = true,
    // @TODO: update tests to use realIntlProvider everywhere
    realIntlProvider = false,
  }: TestProvidersProps): FC<PropsWithChildren<{}>> =>
  ({ children }) => {
    const TranslationsProvider = realIntlProvider ? IntlProvider : I18nProvider;

    return (
      <Provider store={store}>
        <BrowserRouter>
          <ThemeProvider theme={theme}>
            <TranslationsProvider locale="pl" messages={plMessages}>
              <Normalize />
              <GlobalStyles />
              {children}
              {renderModal && <Modal />}
            </TranslationsProvider>
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    );
  };

const customRender = (
  ui: ReactElement,
  options: Omit<RenderOptions, "options"> = {},
  testProviderOptions: TestProvidersProps = {}
) => render(ui, { wrapper: addTestProviders(testProviderOptions), ...options });

export * from "@testing-library/react";

// override render method
export { customRender as render };
