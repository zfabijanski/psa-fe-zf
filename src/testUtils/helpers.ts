export const sleep = (ms: number = 100): Promise<unknown> =>
  new Promise((resolve) => setTimeout(resolve, ms));
