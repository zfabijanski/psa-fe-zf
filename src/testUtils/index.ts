export * from "./TestProviders";
export * from "./mockCreators";
export * from "./helpers";
