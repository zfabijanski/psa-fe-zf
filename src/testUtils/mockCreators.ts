import {
  ICalculatorCoversConfig,
  IProductCovers,
} from "features/Calculator/types";
import {
  DocumentState,
  IDirectory,
  IDirectoryMetadata,
  IDocumentMetadata,
} from "features/Library/types";
import { IMeeting } from "features/Meeting/History/types";
import { IIllustration, ProductGuid } from "features/Products/types";
import { IModalData, ModalTypes } from "models/Modal";
import {
  Frequency,
  IFundRiskProfileConfig,
  IProductConfig,
  IProductFund,
  IProductGuarantee,
  InvestmentProfile,
  Role,
} from "models/common";
import {
  CalculationStatus,
  Direction,
  ICalculationDetailsDto,
  ICover,
  IInsured,
  YesNo,
} from "models/calculator";

export const createCalculationDetailsDto = (
  overrides?: Partial<ICalculationDetailsDto>
): ICalculationDetailsDto => ({
  calculation_id: 1,
  working_date: "2020-01-01",
  premiums_total: 100,
  guaranteed_death_benefit: 100,
  guaranteed_maturity_benefit: 100,
  projected_maturity_value_egr_m: 100,
  product_categories: [],
  uuid: "uuid",
  version: "version",
  source_system: "source_system",
  product_guid: 1,
  insureds_list: [],
  covers_list: [],
  funds_list: [],
  ...overrides,
});

export const createInsured = (overrides?: Partial<IInsured>): IInsured => ({
  insured_idx: 1,
  insured_uuid: "insured_uuid",
  age_in_months: 1,
  date_of_birth: "2020-01-01",
  role: Role.MAIN,
  ...overrides,
});

export const createCover = (overrides?: Partial<ICover>): ICover => ({
  cover_id: 1,
  cover_idx: 1,
  insured_idx: 1,
  code: "code",
  frequency_code: Frequency.Annually,
  payments: 1,
  direction_of_calculation: Direction.SumAssuredToPremium,
  investment_profile: InvestmentProfile.Balanced,
  indexation: YesNo.Yes,
  sum_assured: 1,
  premium_per_freq: 1,
  duration: 1,
  idx_option: YesNo.Yes,
  rop_option: YesNo.Yes,
  add_sum_assured: 1,
  ending_age: 1,
  exp_mat_ben: 1,
  ...overrides,
});

export const createCalculatorCoversConfig = (
  overrides?: Partial<ICalculatorCoversConfig>
): ICalculatorCoversConfig => ({
  coverName: "coverName",
  isChild: false,
  isHospital: false,
  ...overrides,
});

export const createCalculatorCoversConfigMap = (
  overrides?: Partial<Record<string, ICalculatorCoversConfig>>
): Record<string, ICalculatorCoversConfig> => ({
  coverName1: createCalculatorCoversConfig({ coverName: "coverName1" }),
  coverName2: createCalculatorCoversConfig({ coverName: "coverName2" }),
  ...overrides,
});

export const createProductCovers = (
  overrides?: Partial<IProductCovers>
): IProductCovers => ({
  mainCover: "mainCover",
  additionalCovers: ["additionalCover1", "additionalCover2"],
  wopCovers: ["wopCover1", "wopCover2"],
  childCovers: ["childCover1", "childCover2"],
  ...overrides,
});

export const createProductCoversMap = (
  overrides?: Partial<Record<string, IProductCovers>>
): Record<string, IProductCovers> => ({
  mainCover1: createProductCovers({ mainCover: "mainCover1" }),
  mainCover2: createProductCovers({ mainCover: "mainCover2" }),
  ...overrides,
});

export const createProductConfig = (
  overrides?: Partial<IProductConfig>
): IProductConfig => ({
  product_guid: 1,
  product_name: "product_name",
  product_categories: [],
  product_freqs: [],
  product_covers: [],
  product_funds: [],
  ...overrides,
});

export const createProductConfigMap = (
  overrides?: Partial<Record<string, IProductConfig>>
): Record<string, IProductConfig> => ({
  1: createProductConfig({ product_name: "product_name1" }),
  2: createProductConfig({ product_name: "product_name2" }),
  ...overrides,
});

export const createFundRiskProfileConfig = (
  overrides?: Partial<IFundRiskProfileConfig>
): IFundRiskProfileConfig => ({
  code: "code",
  name_pl: "name_pl",
  name_en: "name_en",
  risk_level: 1,
  status_code: "status_code",
  notes: "notes",
  ...overrides,
});

export const createProductGuarantee = (
  overrides?: Partial<IProductGuarantee>
): IProductGuarantee => ({
  guarantee_code: "guarantee_code",
  guarantee_name: "guarantee_name",
  guarantee_min_value: "1",
  guarantee_duration_months: "4",
  guarantee_percentages: [{ guarantee_percent: 10 }],
  ...overrides,
});

export const createProductFund = (
  overrides?: Partial<IProductFund>
): IProductFund => ({
  cover_code: "cover_code",
  fund_code: "fund_code",
  fund_name: "fund_name",
  fund_risk_profile: "code",
  fund_order: 1,
  guarantees: [createProductGuarantee()],
  ...overrides,
});

export const createModalData = (
  overrides: Partial<IModalData> = {}
): IModalData => ({
  modalIllustration: "alert-fill",
  modalTitleTrKey: "modalTitleTrKey",
  modalContentTrKey: "modalContentTrKey",
  modalContent: "modalContent",
  modalButtons: {
    confirm: { textTrKey: "modal.ok" },
  },
  modalType: ModalTypes.default,
  maskClosable: true,
  ...overrides,
});

export const createIllustration = (
  overrides: Partial<IIllustration> = {}
): IIllustration => ({
  idd_id: 1,
  product_guid: ProductGuid.Savings,
  adequacy_id: 1,
  order_id: 1,
  expected_amount: { value: 3 },
  premium_total: { value: 3 },
  duration: { value: 3 },
  premium_main_cover: { value: 3 },
  frequency: { value: 3 },
  sum_of_premiums: { value: 3 },
  sum_assured: { value: 3 },
  investment_profile: { value: 3 },
  pru_calc_calculation_id: 1,
  status: CalculationStatus.Illustrated,
  json_needs_analysis: null,
  ...overrides,
});

export const createDirectory = (
  overrides?: Partial<IDirectory>
): IDirectory => ({
  directoryId: 1,
  directories: [],
  documentsIds: [1, 2, 3],
  ...overrides,
});

export const createDirectoryMetadata = (
  overrides?: Partial<IDirectoryMetadata>
): IDirectoryMetadata => ({
  directoryId: 1,
  name: "name",
  isOpen: false,
  ...overrides,
});

export const createDocumentMetadata = (
  overrides?: Partial<IDocumentMetadata>
): IDocumentMetadata => ({
  documentId: 1,
  guid: "guid",
  name: "name",
  state: DocumentState.NotAttached,
  isStateChanging: false,
  moduleName: "PRU_LIBRARY",
  ...overrides,
});

export const createMeeting = (overrides?: Partial<IMeeting>): IMeeting => ({
  meetingNo: "meetingNo",
  meetingId: 1,
  name: "name",
  age: 18,
  lastAccessDate: new Date("2023-01-01"),
  createDate: new Date("2023-01-01"),
  statusId: 1,
  statusName: "statusName",
  statusDescription: "statusDescription",
  state: "state",
  haveAdequacy: false,
  haveCalculation: false,
  haveIdd: false,
});
