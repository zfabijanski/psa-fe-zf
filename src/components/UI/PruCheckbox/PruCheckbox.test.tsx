import { PruCheckbox, ECheckboxType, ICheckboxProps } from "./PruCheckbox";
import { render, screen } from "testUtils";

const mockOnChange = jest.fn();

const defaultProps: ICheckboxProps = {
  onChange: mockOnChange,
};

const setup = (props: Partial<ICheckboxProps> = {}) =>
  render(<PruCheckbox {...defaultProps} {...props} />);

describe("PruCheckbox", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it("should call onChange when checkbox is clicked", () => {
    setup();
    screen.getByRole("checkbox").click();
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });

  it("should handle different types", () => {
    setup({ checkboxType: "regular" });
    expect(screen.getByRole("checkbox")).toBeInTheDocument();
  });

  it("should render the label", () => {
    setup({ labelProps: { labelText: "label" } });
    expect(screen.getByText("label")).toBeInTheDocument();
  });

  it.each([
    "regular",
    "gridItem",
    "calcTableItem",
    "checkboxList",
  ] as ECheckboxType[])(
    `should render the %s checkbox`,
    (checkboxType: ECheckboxType) => {
      setup({ checkboxType });
      expect(screen.getByRole("checkbox")).toMatchSnapshot();
    }
  );

  it("should handle i18n in label", () => {
    setup({ labelProps: { labelTrKey: "label-translation-key" } });
    expect(screen.getByText("label-translation-key")).toBeInTheDocument();
  });

  it("should handle invalid case", () => {
    setup({ isInvalid: true });
    expect(screen.getByRole("checkbox")).toHaveStyle("border-color: #d13c15");
  });

  it("should handle disabled case", () => {
    setup({ disabled: true });
    expect(screen.getByRole("checkbox")).toHaveStyle("border-color: #999999");
  });
});
