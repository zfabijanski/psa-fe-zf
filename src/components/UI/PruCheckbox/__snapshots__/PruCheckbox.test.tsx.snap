// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PruCheckbox should render correctly 1`] = `
.c4 {
  display: inline-block;
  cursor: inherit;
}

.c4 > svg {
  width: 16px;
  height: 100%;
  display: block;
}

.c4 svg g,
.c4 svg g path,
.c4 svg g circle,
.c4 svg g rect {
  stroke: #FFFFFF !important;
}

.c4 svg g rect {
  fill: #FFFFFF !important;
}

.c4 svg g tspan {
  stroke: transparent !important;
  fill: #FFFFFF !important;
}

.c4 svg > path {
  stroke: #FFFFFF !important;
}

.c8 {
  font-weight: 600;
  font-size: 14px;
  line-height: 20px;
}

.c7 {
  margin-top: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c9 {
  color: #D13C15;
  line-height: 18px;
  min-height: 18px;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c3 {
  width: 24px;
  min-width: 24px;
  height: 24px;
  border: 1px solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 12px;
  cursor: pointer;
  background-color: #F9FDFD;
  border-color: #17B0AD;
}

.c3:before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c3:hover {
  background-color: #DCF3F3;
  border-color: #055A60;
}

.c6 {
  visibility: hidden;
}

.c1 {
  display: none;
}

.c1:checked + .c2 .c5 {
  visibility: initial;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      type="checkbox"
    />
    <div
      class="c2 c3"
      role="checkbox"
    >
      <div
        class="c4 c5 c6"
        color="white100"
      >
        <svg>
          check-white.svg
        </svg>
      </div>
    </div>
  </div>
  <div
    class="c7"
  >
    <div
      class="c8 c9"
    />
  </div>
  <div />
</div>
`;

exports[`PruCheckbox should render the calcTableItem checkbox 1`] = `
.c2 {
  display: inline-block;
  cursor: inherit;
}

.c2 > svg {
  width: 16px;
  height: 100%;
  display: block;
}

.c2 svg g,
.c2 svg g path,
.c2 svg g circle,
.c2 svg g rect {
  stroke: #FFFFFF !important;
}

.c2 svg g rect {
  fill: #FFFFFF !important;
}

.c2 svg g tspan {
  stroke: transparent !important;
  fill: #FFFFFF !important;
}

.c2 svg > path {
  stroke: #FFFFFF !important;
}

.c1 {
  width: 24px;
  min-width: 24px;
  height: 24px;
  border: 1px solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 12px;
  cursor: pointer;
  background-color: #F9FDFD;
  border-color: #17B0AD;
}

.c1:before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c1:hover {
  background-color: #DCF3F3;
  border-color: #055A60;
}

.c4 {
  visibility: hidden;
}

.c5:checked + .c0 .c3 {
  visibility: initial;
}

<div
  class="c0 c1"
  role="checkbox"
>
  <div
    class="c2 c3 c4"
    color="white100"
  >
    <svg>
      check-white.svg
    </svg>
  </div>
</div>
`;

exports[`PruCheckbox should render the checkboxList checkbox 1`] = `
.c2 {
  display: inline-block;
  cursor: inherit;
}

.c2 > svg {
  width: 16px;
  height: 100%;
  display: block;
}

.c2 svg g,
.c2 svg g path,
.c2 svg g circle,
.c2 svg g rect {
  stroke: #FFFFFF !important;
}

.c2 svg g rect {
  fill: #FFFFFF !important;
}

.c2 svg g tspan {
  stroke: transparent !important;
  fill: #FFFFFF !important;
}

.c2 svg > path {
  stroke: #FFFFFF !important;
}

.c1 {
  width: 24px;
  min-width: 24px;
  height: 24px;
  border: 1px solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 12px;
  cursor: pointer;
  background-color: #F9FDFD;
  border-color: #17B0AD;
}

.c1:before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c1:hover {
  background-color: #DCF3F3;
  border-color: #055A60;
}

.c4 {
  visibility: hidden;
}

.c5:checked + .c0 .c3 {
  visibility: initial;
}

<div
  class="c0 c1"
  role="checkbox"
>
  <div
    class="c2 c3 c4"
    color="white100"
  >
    <svg>
      check-white.svg
    </svg>
  </div>
</div>
`;

exports[`PruCheckbox should render the gridItem checkbox 1`] = `
.c2 {
  display: inline-block;
  cursor: inherit;
}

.c2 > svg {
  width: 16px;
  height: 100%;
  display: block;
}

.c2 svg g,
.c2 svg g path,
.c2 svg g circle,
.c2 svg g rect {
  stroke: #FFFFFF !important;
}

.c2 svg g rect {
  fill: #FFFFFF !important;
}

.c2 svg g tspan {
  stroke: transparent !important;
  fill: #FFFFFF !important;
}

.c2 svg > path {
  stroke: #FFFFFF !important;
}

.c1 {
  width: 24px;
  min-width: 24px;
  height: 24px;
  border: 1px solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 12px;
  cursor: pointer;
  background-color: #F9FDFD;
  border-color: #17B0AD;
}

.c1:before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c1:hover {
  background-color: #DCF3F3;
  border-color: #055A60;
}

.c4 {
  visibility: hidden;
}

.c5:checked + .c0 .c3 {
  visibility: initial;
}

<div
  class="c0 c1"
  role="checkbox"
>
  <div
    class="c2 c3 c4"
    color="white100"
  >
    <svg>
      check-white.svg
    </svg>
  </div>
</div>
`;

exports[`PruCheckbox should render the regular checkbox 1`] = `
.c2 {
  display: inline-block;
  cursor: inherit;
}

.c2 > svg {
  width: 16px;
  height: 100%;
  display: block;
}

.c2 svg g,
.c2 svg g path,
.c2 svg g circle,
.c2 svg g rect {
  stroke: #FFFFFF !important;
}

.c2 svg g rect {
  fill: #FFFFFF !important;
}

.c2 svg g tspan {
  stroke: transparent !important;
  fill: #FFFFFF !important;
}

.c2 svg > path {
  stroke: #FFFFFF !important;
}

.c1 {
  width: 24px;
  min-width: 24px;
  height: 24px;
  border: 1px solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 12px;
  cursor: pointer;
  background-color: #F9FDFD;
  border-color: #17B0AD;
}

.c1:before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c1:hover {
  background-color: #DCF3F3;
  border-color: #055A60;
}

.c4 {
  visibility: hidden;
}

.c5:checked + .c0 .c3 {
  visibility: initial;
}

<div
  class="c0 c1"
  role="checkbox"
>
  <div
    class="c2 c3 c4"
    color="white100"
  >
    <svg>
      check-white.svg
    </svg>
  </div>
</div>
`;
